metadata.md

```diff
@@ -0,0 +1,212 @@
+                                 'type': 'string'},
+                   'pcr': {'description': 'Was PCR used on this sample',
+                           'name': 'PCR Used',
+                           'type': 'boolean'}},
+    'required': ['accession', 'pcr'],
+    'type': 'object',
+})
+```
+
+This {class}`MetadataSchema` can then be assigned to a table or the top-level
+tree sequence e.g. {attr}`~IndividualTable.metadata_schema`:
+
+```{code-cell} ipython3
+tables = tskit.TableCollection(sequence_length=1)  # make a new, empty set of tables
+tables.individuals.metadata_schema = complex_schema
+```
+
+This will overwrite any existing schema. Note that this will not validate any existing
+metadata against the new schema. Now that the table has a schema, calls to
+{meth}`~IndividualTable.add_row` will validate and encode the metadata:
+
+```{code-cell} ipython3
+tables.individuals.add_row(0, metadata={"accession": "Bob1234", "pcr": True})
+```
+
+If we try to add metadata that doesn't fit the schema, such as accidentally using a
+string instead of a proper Python boolean, we'll get an error:
+
+```{code-cell} ipython3
+:tags: [raises-exception, output_scroll]
+tables.individuals.add_row(0, metadata={"accession": "Bob1234", "pcr": "false"})
+```
+
+and because we set ``additionalProperties`` to ``False`` in the schema, an error is
+also raised if we attempt to add new fields:
+
+```{code-cell} ipython3
+:tags: [raises-exception, output_scroll]
+tables.individuals.add_row(0, metadata={"accession": "Bob1234", "pcr": True, "newKey": 25})
+```
+
+
+To set the top-level metadata, just assign it. Validation and encoding happen as specified by the
+top-level metadata schema (Note that provenance info should go in {ref}`sec_provenance`
+not metadata):
+
+```{code-cell} ipython3
+tables.metadata_schema = basic_schema  # Allows new fields to be added that are not validated
+tables.metadata = {"mean_coverage": 200.5}
+```
+
+To modify a schema, for example to add a key; first get the dict representation, modify
+then write back:
+
+```{code-cell} ipython3
+schema_dict = tables.individuals.metadata_schema.schema
+schema_dict["properties"]["newKey"] = {"type": "integer"}
+tables.individuals.metadata_schema = tskit.MetadataSchema(schema_dict)
+# Now this will work:
+tables.individuals.add_row(metadata={'accession': 'abc123', 'pcr': False, 'newKey': 25})
+```
+
+To modify the metadata of rows in tables use the {ref}`sec_tutorial_metadata_bulk`.
+
+## Viewing raw metadata
+
+If you need to see the raw (i.e., bytes) metadata, you just need to remove the
+schema, for instance:
+
+```{code-cell} ipython3
+print(tables.individuals[0].metadata)
+
+tables.individuals.MetadataSchema = tskit.MetadataSchema(None)
+print(tables.individuals[0].metadata)
+```
+
+(sec_tutorial_metadata_bulk)=
+
+## Metadata for bulk table methods
+
+In the interests of efficiency each table's {meth}`~NodeTable.packset_metadata` method,
```

Comment:
Some link code here is broken - "{meth}" is in the output.

Reply:
Missing backpack. Corrected